File: Muc_Do_1_Co_Ban/Bai3_ChiSoCanBang/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    if (fopen("BAI3.INP", "r")) {
        freopen("BAI3.INP", "r", stdin);
        freopen("BAI3.OUT", "w", stdout);
    }
    int n, q;
    if (!(cin >> n >> q)) return 0;
    vector<long long> prefXor(n + 1, 0);
    for (int i = 1; i <= n; ++i) {
        long long x;
        cin >> x;
        prefXor[i] = prefXor[i-1] ^ x;
    }
    while (q--) {
        int l, r;
        cin >> l >> r;
        cout << (prefXor[r] ^ prefXor[l-1]) << "\n";
    }
    return 0;
}

File: Muc_Do_1_Co_Ban/Bai3_ChiSoCanBang/Solution.py
import sys
def solve():
    try:
        sys.stdin = open('BAI3.INP', 'r')
        sys.stdout = open('BAI3.OUT', 'w')
    except:
        pass
    data = sys.stdin.read().split()
    if not data: return
    n = int(data[0])
    q = int(data[1])
    arr = [int(x) for x in data[2:2+n]]
    pref = [0] * (n + 1)
    for i in range(n):
        pref[i+1] = pref[i] ^ arr[i]
    idx = 2 + n
    out = []
    for _ in range(q):
        l = int(data[idx])
        r = int(data[idx+1])
        idx += 2
        out.append(str(pref[r] ^ pref[l-1]))
    print('\n'.join(out))
solve()
